inspirehep/modules/workflows/tasks/refextract.py

```diff
@@ -86,49 +87,45 @@ def extract_journal_info(obj, eng):
 
 def extract_references(filepath):
     """Extract references from PDF and return in INSPIRE format."""
+
+    def field_builder(builder):
+        def _build_field(value_l):
+            for value in value_l:
+                builder(value)
+        return _build_field
```

Comment:
This high order function is the partial application of `map` to `builder`. It should be possible to rewrite it in terms of `functools.partial` and `map`.